common/cache/lru.go

```
@@ -319,13 +320,14 @@ func (c *lru) evictExpiredItems() {
 // allowUpdate flag is used to control overwrite behavior if the value exists
 func (c *lru) putInternal(key interface{}, value interface{}, allowUpdate bool) (interface{}, error) {
 	valueSize := uint64(0)
-	if c.isSizeBased {
-		sizeableValue, ok := value.(Sizeable)
-		if !ok {
-			return nil, fmt.Errorf("value %T does not implement sizable. Key: %+v", value, key)
-		}
+	sizeableValue, ok := value.(Sizeable)
+	if !ok {
+		c.isSizeBased = dynamicproperties.GetBoolPropertyFn(false)
```

I prefer to return an error in this case. I'd like to have the cache behavior to be completely controlled by dynamic config so that it's easier to understand the behavior and debug issues..

---

I add this condition because currently many usage of the cache do not implement Sizeable interface (not having the ByteSize() method). I want to enable the size-based cache for those usage incrementally while keeping the current logic intact. Returning an error in this case would cause issue for many cache used outside of cadence history.

---

We can define separate dynamicconfig properties for different cache usages and enabled them one by one. Supporting fallbacks based on `value`'s type might sound safe but it makes it hard to reason about the system as it introduces more complexity. 
Let's aim for explicit and simple behavior in this cache layer. We may move this cache package to its own repo (a standalone OS repo) eventually.

---

Sounds good, in current state where not all cache entry types are not implementing the ByteSize() method, I will emit a warning log instead?